storage/spanstore/downsampling_writer_benchmark_test.go

```diff
@@ -0,0 +1,81 @@
+package spanstore
+
+import (
+	"math"
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/uber/jaeger-lib/metrics"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/stretchr/testify/assert"
+)
+
+type noopWriteSpanStore struct{}
+
+func (n *noopWriteSpanStore) WriteSpan(span *model.Span) error {
+	return nil
+}
+
+//Benchmark result:
+//BenchmarkDownSamplingWriter_WriteSpan-12 without bytepool 50	  33352181 ns/op	18981065 B/op	  219750 allocs/op
+//BenchmarkDownSamplingWriter_WriteSpan2-12   with bytepool 100	  28070245 ns/op	16137975 B/op	  127494 allocs/op
+func BenchmarkDownSamplingWriter_WriteSpan(b *testing.B) {
+	trace := model.TraceID{
+		Low:  uint64(0),
+		High: uint64(1),
+	}
+	span := &model.Span{
+		TraceID: trace,
+	}
+	c := NewDownSamplingWriter(&noopWriteSpanStore{}, DownSamplingOptions{
+		Ratio:    0.5,
+		HashSalt: "jaeger-test",
+	})
+	var wg sync.WaitGroup
+	numGoroutines := 10000
+	writeTimes := 10
+	wg.Add(numGoroutines * b.N)
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	for j := 0; j < b.N; j++ {
+		for n := 0; n < numGoroutines; n++ {
+
+			go func() {
```

Comment:
this probably allocates memory every time. Try moving outside of the loop and assigning to a variable, e.g. `f := func()...`, then inside the loop only call `f()`

Reply:
actually, you should avoid creating go-routing inside the loop after you called `b.ResetTimer()`. You can start then all before `b.ResetTimer()`, and have them all block on reading from a channel. Then close the channel to unblock them all.

Reply:
I've removed the goroutines entirely from benchmark test for now.